### PR TITLE
Adds layer's visible parameter

### LINKED ISSUE #543: Layer "visible" property should set layer visibility
https://github.com/tangrams/tangram-es/blob/master/core/src/scene/sceneLoader.cpp#L1354

https://github.com/tangrams/tangram-docs/blob/gh-pages/pages/layers.md#visible

### PR DESCRIPTION
**Pull request by tallytalwar**
Fixes #543 

Added layer:visible property and respect skip drawing for layer if its marked not visible.

### CODE REVIEW

File: core/src/scene/sceneLayer.cpp [hunk 1]
```diff
@@ -6,11 +6,13 @@ namespace Tangram {
 
 SceneLayer::SceneLayer(std::string _name, Filter _filter,
                        std::vector<DrawRuleData> _rules,
-                       std::vector<SceneLayer> _sublayers) :
+                       std::vector<SceneLayer> _sublayers,
+                       bool _visible) :
```
Comment (karimnaaji) [inline core/src/scene/sceneLayer.cpp]: Can we default the value to `true` in the constructor instead?

**Comment by karimnaaji** [comment]
Good to go once filter constructor is updated with default value or unit tests adapted!

**Comment by hjanetzek** [comment]
Why do we add invisible layers at all and don't throw them out on load?

**Comment by karimnaaji** [comment]
I guess this could be a dynamically changed property for debug or other purpose.

**Comment by tallytalwar** [comment]
I did think about throwing them out on load,  but if we provide hooks for
dynamic changing properties then we need to have these in scene.

**Comment by bcamper** [comment]
Yeah, to ensure compatibility with JS I don't think we should get too
clever with throwing them out for now.

File: core/src/scene/sceneLayer.cpp (hunk 1 above)
Comment (tallytalwar) [inline core/src/scene/sceneLayer.cpp]: :+1:

**Pull request merged**